src/v/cluster/archival_metadata_stm.cc

```diff
@@ -664,6 +800,8 @@ ss::future<std::error_code> archival_metadata_stm::do_replicate_commands(
         co_return errc::replication_error;
     }
 
+    _last_replicated_offset = batch_last_offset;
```

Comment:
I'm not quite following why do we need this. At this point the `batch_last_offset` is already applied to the STM. The `wait_no_throw` is waiting until the offset in first argument is applied to the STM. So the `_last_replicated_offset` should be equal to `_insync_offset`.

Reply:
You're correct. This is misplaced. It should be updated after the `replicate` call. Will move.

Reply:
fixed